src/screens/OnboardingScreen.tsx

```diff
@@ -192,9 +222,14 @@ export const OnboardingScreen = ({
             <Button
               style={styles.completeButton}
               type="primary"
-              onPress={() => {
+              onPress={async () => {
                 if (isLastOnboardingPage) {
-                  onComplete();
+                  await getTrackingPermission();
+                  if (!backgroundPermission.granted) {
+                    requestPermissions();
+                  } else {
+                    onComplete();
+                  }
```

Comment:
This whole thing should also be a function I think, just cleaner than leaving this in the button code. Also this needs error catching.